cmd/state-offline-installer/install.go

```diff
@@ -89,62 +116,144 @@ func (r *runner) Run(params *Params) error {
 		if err != nil {
-			return errs.Wrap(err, "Unable to open License file")
+			return r.handleFailure(err, "Unable to open License file", installerDimensions)
 		}
 
 		accepted, err := legalprompt.CustomLicense(string(b), r.out, r.prompt)
 		if err != nil {
-			return errs.Wrap(err, "Error with license acceptance")
+			return r.handleFailure(err, "Error with license acceptance", installerDimensions)
 		}
 		if !accepted {
-			return locale.NewInputError("License not accepted")
+			return r.handleAbort(
+				locale.NewInputError("License not accepted"),
+				"License failure",
+				installerDimensions,
+			)
 		}
 	}
 
 	/* Extract Artifacts */
 	artifactsPath := filepath.Join(tempDir, artifactsPathName)
 	if err := r.extractArtifacts(artifactsPath, assetsPath); err != nil {
-		return errs.Wrap(err, "Could not extract artifacts")
+		return r.handleFailure(err, "Could not extract artifacts", installerDimensions)
 	}
 
 	/* Install Artifacts */
 	offlineTarget := target.NewOfflineTarget(params.path, artifactsPath)
-	asrt, err := r.setupRuntime(artifactsPath, offlineTarget)
+	asrt, err := r.setupRuntime(artifactsPath, offlineTarget, logfile)
 	if err != nil {
-		return errs.Wrap(err, "Could not setup runtime")
+		return r.handleFailure(err, "Could not setup runtime", installerDimensions)
 	}
 
 	/* Manually Install License File */
 	{
 		err = fileutils.CopyFile(licenseFileAssetPath, filepath.Join(params.path, licenseFileName))
 		if err != nil {
-			return errs.Wrap(err, "Error copying license file")
+			return r.handleFailure(err, "Error copying license file", installerDimensions)
+		}
+	}
+
+	/* Manually Install config File */
+	{
+		err = fileutils.CopyFile(
+			installerConfigPath,
+			filepath.Join(params.path, installerConfigFileName),
+		)
+		if err != nil {
+			return r.handleFailure(err, "Error copying config file", installerDimensions)
+		}
+	}
+
+	var uninstallerSrc string
+	var uninstallerDest string
+
+	/* Manually Install uninstaller */
+	if rt.GOOS == "windows" {
+		/* shenanigans because windows won't let you delete an executable that's running */
+		installDir, err := filepath.Abs(params.path)
+		if err != nil {
+			return r.handleFailure(err, "Error determining absolute install directory", installerDimensions)
+		}
+		uninstallDir := filepath.Join(installDir, "uninstall-data")
+		if err := os.Mkdir(uninstallDir, os.ModeDir); err != nil {
+			return r.handleFailure(err, "Error creating uninstall directory", installerDimensions)
+		}
+		uninstallerDestName := fmt.Sprintf("%s-%s-%s.exe", *config.ProjectID, *config.CommitID, uninstallerFileNameRoot)
+
+		uninstallerSrc = filepath.Join(assetsPath, uninstallerFileNameRoot+".exe")
+		uninstallerDest = filepath.Join(uninstallDir, uninstallerDestName)
+
+		// create batch script
+		batch := fmt.Sprintf(
+			"@echo off\ncopy %s\\%s %%TEMP%%\\%s >nul 2>&1\n%%TEMP%%\\%s %s & del %%TEMP%%\\%s >nul 2>&1 & echo You can safely ignore the following error message:\n",
```

Comment:
Can we split this into its own bat file (you can use https://pkg.go.dev/embed to bring it in), and make it a bit more verbose (whitespace, new lines, comments)?

Seeing as the intend is for DevX to eventually take over I'd want this to be in a more maintainable form. This takes a lot of effort to unpack currently.

Reply:
For your consideration, it might even be better to reuse this code:

https://github.com/ActiveState/cli/blob/65113461adf92c91713c04c92b2782d0c04995be/internal/runners/clean/run_win.go#L120

Which uses this batch file:

https://github.com/ActiveState/cli/blob/904d0f66f51ff2a91bfb7b33ffeebdb3fdc2a0e0/internal/assets/contents/scripts/removePaths.bat#L1-L0